### ISSUE TITLE
Deep linking issues

### ISSUE THREAD

**Issue by siepra** [TryQuiet/quiet#2253]
We're experiencing several problems sharing the same origin which is abnormal behaviour triggered by using deep linking.
In order to make debugging easier, I decided to merge independently reported issues into the single task.

Overall reproduction steps presents as follows:
1. Open the app
2. Pass the network creation phase (eg. by reaching username registration)
3. Use deep linking feature (open the app from url)
4. (optional) Leave a community before executing step no. 3

(Draft) PR
https://github.com/TryQuiet/quiet/pull/2154

Originally reported task:
https://github.com/TryQuiet/quiet/pull/1970

**Comment by siepra** [comment]
Another important observation:
Deep linking issues have nothing to do with leaving community. If one leave community and then rejoin pasting invite link instead of using deep linking feature, it works correctly.
On the other hand, it's enough to just open the app (and wait for state-manager sagas to be forked) and then use deep linking feature to mess up the app.

**Comment by kingalg** [comment]
On `fix/1970` branch Android seems to work fine, so the problem is likely iOS specific.

I will check it again (on both Android and iOS) once the mentioned branch is merged into develop and released with the alpha version. I'll also check for any other abnormalities. 

We agreed with @siepra that any spotted issues will be reported as new tasks, no sooner than `fix/1970` is merged, to avoid chaos and confusion.

**Comment by kingalg** [comment]
mobile@2.1.1-alpha.0 | iOS 350

Fixed.
